web/packages/teleport/src/Main/Main.tsx

```diff
@@ -189,19 +197,31 @@ export function Main(props: MainProps) {
       <Wrapper>
         <MainContainer>
           <Navigation />
-          <ContentWrapper>
-            <ContentMinWidth>
-              <BannerList
-                banners={banners}
-                customBanners={props.customBanners}
-                billingBanners={featureFlags.billing && props.billingBanners}
-                onBannerDismiss={dismissAlert}
-              />
-              <Suspense fallback={null}>
-                <FeatureRoutes lockedFeatures={ctx.lockedFeatures} />
-              </Suspense>
-            </ContentMinWidth>
-          </ContentWrapper>
+          <InfoGuidePanelContext.Provider
+            value={{ infoGuideElement, setInfoGuideElement }}
```

Comment:
This seems a bit unorthodox to me because we define some state in `Main` and then pass the state and the setter to a context provider. Wouldn't it be more idiomatic to define the state and the setters in the context provider itself and then use the context inside `ContentMinWidth` and `InfoGuideSidePanel`? I suppose you'd probably want to have some kind of a wrapper for `InfoGuideSidePanel` that reads this stuff from the context.

Reply:
thanks, I created a InfoGuideSidePanelProvider